source/elements/oneDAL/source/data_management/tables.rst

```diff
@@ -1,3 +1,290 @@
+
+- Are dense and stored as one continuous memory block
+- All features have the same data type (but feature types may differ)
+
+::
+
+   class homogen_table : public table {
+   public:
+      homogen_table(const homogen_table&);
+      homogen_table(homogen_table&&);
+
+      homogen_table(int64_t N, int64_t p, data_layout layout);
+
+      template <typename T>
+      homogen_table(T* data_pointer, int64_t N, int64_t p, data_layout layout);
+
+      homogen_table& operator=(const homogen_table&);
+
+      data_type get_data_type() const noexcept;
+      bool is_feature_types_equal() const noexcept;
+
+      template <typename T>
+      const T* get_data_pointer() const noexcept;
+   };
+
+data_type ``data_type`` ``[read-only]``
+   The actual type of underlying data
+
+feature_types_equal ``bool``
+   Flag indication whether the `feature_type` fields
+   of `metadata` are all equal
+
+data_pointer ``const T*``
+   The pointer to underlying data
+
+.. _soa_table:
+
+Structure-of-arrays table
+-------------------------
+TBD
+
+.. _aos_table:
+
+Arrays-of-structure table
+-------------------------
+TBD
+
+.. _csr_table:
+
+Compressed-sparse-row table
+---------------------------
+TBD
+
+.. _Metadata API:
+
+------------
+Metadata API
+------------
+Table metadata contains structures describing how the data
+are stored inside the table and how efficiently access them.
+
+::
+
+   class table_meta {
+   public:
+      table_meta();
+
+      int64_t get_feature_count() const noexcept;
+      table_meta& set_features_count(int64_t);
+
+      const feature& get_feature(std::int64_t index) const;
+      table_meta& add_feature(const feature&);
+
+      data_layout get_layout() const noexcept;
+      table_meta& set_layout(data_layout);
+
+      bool is_continuous() const noexcept;
+      table_meta& set_continuous(bool);
+
+      bool is_homogeneous() const noexcept;
+
+      data_format get_format() const noexcept;
+      table_meta& set_format(data_format);
+   };
+
+feature_count ``int64_t`` ``default = 0``
+   The number of features :math:`p` in the table.
+
+   Invariants
+      | ``feature_count >= 0``
+
+feature ``feature``
+   An info about particular feature in the table
+
+layout ``data_layout`` ``default = data_layout::row_major``
+   Flag indicating whether the data are in C or Fortran format.
+
+is_continuous ``bool`` ``default=true``
+   Indicates whether the data are stored in continuous block of memory.
+   E.g., if ``is_continuous == true`` and ``data_layout`` is ``row_major``,
+   the data are stored continuously row-by-row.
+
+is_homogeneous ``bool`` ``default=true``
+   True if all features has the same ``data_type``
+
+format ``data_format`` ``default=dense``
+   Description of format used for data representation inside the table
+
+Feature
+-------
+::
+
+   class feature {
+   public:
+      dal::data_type get_data_type() const noexcept;
+      dal::feature_type get_type() const noexcept;
+   };
+
+Data layout
```

Comment:
just in case, we start using data layout and data format terms and make silent assumption the reader would have similar to our definition. I suggest to 1) define the terms in terminology, 2) double-check and confirm the defs are accurate and are used by the community in the same sense